src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -1023,35 +981,69 @@ func (r *ClusterReconciler) doDecommissionGhostBrokers(c context.Context, vClust
 	if err != nil {
 		return fmt.Errorf("unable to create admin client: %w", err)
 	}
-	adminBrokers, err := adminClient.Brokers(ctx)
+
+	health, err := adminClient.GetHealthOverview(ctx)
+	if err != nil {
+		return fmt.Errorf("feating cluster heatlh overview: %w", err)
+	}
+
+	// Until cluster does not report that its healthy do not decommission
+	if !health.IsHealthy {
+		err = errors.New("cluster is not healthy") //nolint:goerr113,err113 // That error will not be handled by reconciler function
+		log.V(logger.DebugLevel).Error(err, "stopping decommission verification",
```

Comment:
Similar question to do we want to return an error? This feels more like a warning or info log line.

Also I don't think "stopping decommission verification" is very descriptive here. How about "skipping ghost broker checks due to cluster health"?

Reply:
> Also I don't think "stopping decommission verification" is very descriptive here. How about "skipping ghost broker checks due to cluster health"?

Agree will change.

> Similar question to do we want to return an error? This feels more like a warning or info log line.

I will change it to warn log line without an error.